### PR TITLE
WIP: Feature/custom tests

### PR DESCRIPTION
**Pull request by PWrzesinski**
This a work in progress pull request. What I need is to add screenshot testing to my standard UI tests. Currently out of the box PixelTest can be added to unit tests only. This PR attempts to:

1. Add support for UI tests.
2. Add custom tests API, which a developer can use for any number of purposes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,10 +118,16 @@ The PixelTest example app has examples for how to do it, but with cells specific
 
 PixelTest currently [only works in iOS projects](https://github.com/KaneCheshire/PixelTest/issues/13).
 
+## Changes from the original repository
+
+Enabled PixelTest to be used directly from UI tests.
+
```
Comment (PWrzesinski) [inline README.md]: This needs to be changed of course

**Comment by KaneCheshire** [comment]
Thanks for contributing! I’ll take a look when I get some time.

**Comment by PWrzesinski** [comment]
@KaneCheshire Please note it's still a work in progress 😉

File: PixelTest/Coordinators/FileCoordinator.swift [hunk 2]
```diff
@@ -54,6 +54,19 @@ struct FileCoordinator: FileCoordinatorType {
...
+                     file: StaticString,
+                     scale: Scale,
+                     imageType: ImageType,
+                     layoutStyle: LayoutStyle) -> Bool {
+        let url = fileURL(for: function,
+                          file: file,
+                          scale: scale,
+                          imageType: imageType,
+                          layoutStyle: layoutStyle)
+        return fileManager.fileExists(atPath: url.absoluteString)
```
Comment (PWrzesinski) [inline PixelTest/Coordinators/FileCoordinator.swift]:
```suggestion
        return fileManager.fileExists(atPath: url.relativePath)
```
Just tested

**Comment by KaneCheshire** [comment]
There seems to be a _lot_ of changes going on in this PR which will make it tough to review robustly. Is there a way this could have been broken down into multiple PRs? Incremental changes are always better than one big change

**Comment by PWrzesinski** [comment]
Hey, this is not a big PR, but I'll be happy to break it down if it will make it easier to review it.

File: PixelTest/Coordinators/FileCoordinator.swift [hunk 3]
```diff
@@ -25,8 +25,8 @@ struct FileCoordinator: FileCoordinatorType {
     // MARK: - Functions -
     // MARK: Internal
     
-    /// Attempts to create a full file URL.
-    /// Ensure PIXELTEST_BASE_DIR is set in your scheme before calling this function.
+    /// Creates a full file URL.
+    /// Ensure PIXELTEST_BASE_DIR is set in your scheme before calling this function. // TODO: is this really needed?
```
Comment (PWrzesinski) [inline PixelTest/Coordinators/FileCoordinator.swift:29]: @KaneCheshire is PIXELTEST_BASE_DIR still used or can I remove it?

**Comment by PWrzesinski** [comment]
BTW I'm doing the changes to fit the project I'm currently working on, feel free to completely reject them.

**Pull request closed without merging**